### ISSUE TITLE
request.resource update to request.data for zjs-servers samples

### ISSUE THREAD

**Issue by qiuzhong** [01org/SmartHome-Demo#169]
According to the latest iotivity-constrained OCF specification in [zephyr.js/docs/ocf.md](https://github.com/01org/zephyr.js/blob/master/docs/ocf.md#ocf-server), the OCF `Request` object is defined as: 
```
interface Request {
      OCFResource target;       // Target/destination resource
      OCFResource source;       // Source/origin resource
      object data;              // resource representation
      Promise<void> respond(object data);
  };
```
Thus, `request.resource` which is used in all the zjs-servers examples, now is undefined. All zjs-servers examples are not response the `POST` requests correctly.
```
$ ./oic-get /a/led?di=0acb6309-61cb-4309-72cb-630903cc6309
localhost:8000 /a/led?di=0acb6309-61cb-4309-72cb-630903cc6309
{"value":false}
HTTP: 200
$ cat led_on.json 
{"value": true}
$ ./oic-post /a/led?di=0acb6309-61cb-4309-72cb-630903cc6309 led_on.json 
localhost:8000 /a/led?di=0acb6309-61cb-4309-72cb-630903cc6309
Resource not found.
HTTP: 404
```
We should use `request.data` to sync with API doc.

**Comment by ttzeng** [comment]
@qiuzhong thanks for the notification, I'll update the code and PR accordingly.
:+1:

**Comment by ttzeng** [comment]
@qiuzhong I've updated the led, multiserver, and multiservers sample code with PR #170 to compliant with the new Request API, however due to [ZJS issue #1428](https://github.com/01org/zephyr.js/issues/1428), I am unable to validate the result before submitting the PR. We may wait the resolution from the ZJS project to unblock the verification, by now only led and temperature (single resource) have been validated. You may take a look. Thanks.

**Comment by qiuzhong** [comment]
@ttzeng Yes. I also tried the led.js and temperature.js with new Request API and verified that they can work well.

For multi-sensor cases, due to [ZJS issue #1428](https://github.com/01org/zephyr.js/issues/1428), they still don't work. So we have no way to complete the verification right now. 

Thanks for you quick reply and update!

**Comment by qiuzhong** [comment]
**Update**: [ZJS issue#1428](https://github.com/01org/zephyr.js/issues/1428) had been fixed. All the zjs-server samples can work well with the upstream fix now. I think this issue can be closed now.

**Comment by ttzeng** [comment]
@qiuzhong how many GETs did you test with the multiserver? The board will hung in 5~10 GETs in my test with [ddc30dc](https://github.com/01org/zephyr.js/commit/ddc30dc9db3963977eac933be148ffc54c23aec2)

**Comment by qiuzhong** [comment]
@ttzeng , for multiserver, I can get all the sensors.

**Comment by ttzeng** [comment]
@qiuzhong yes, all servers are discovered from the /api/oic/res endpoint, but once I get the di of a specific server, I can only read the property of that server 6~8 times, then the system hang. Did you observe the symptom at your side?

**Comment by qiuzhong** [comment]
@ttzeng , the behavior you described above also happened at my side! Then I found all the sensors are gone via `/res` REST request. The 6lowpan is still on though. So is is this an upstream issue?

Besides that, I found sometimes the **ZJS Device** cannot be found after the board boots. If I flash the bluetooth firmware and JS image again, the **ZJS Device** might be found.

**Comment by ttzeng** [comment]
@qiuzhong though I didn't check whether the 'ZJS Device' Bluetooth device name has been advertised, I can always connect to the board as I always use fixed Bluetooth address. Anyway, we may need to file an upstream bug for this new symptom.

**Comment by uhofemei** [comment]
I can see the sensors, but using the smart home cloud app to access the sensors via the REST API  will render the sensors unavailable within seconds.

**Comment by wanghongjuan** [comment]
@ttzeng I can reproduce the symptom that you mentioned, and file a new issue https://github.com/01org/zephyr.js/issues/1457, I think we can trick the new issue and close this one now, after all the request.resource has been updated to request.data, and it works.
@uhofemei About the symptom what you said is doesn't matter with this issue, I think we can open a new issue to track it. Is it OK?

**Comment by ttzeng** [comment]
This issue is regarding the change of the Request API, I agree with @qiuzhong we should close this now as there already have open issues tracking the instability.
